modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue

```diff
@@ -89,15 +97,53 @@ export default {
           required: true
         }
       ],
-      context: {}
+      requirements: getRequirements(),
+      context: {},
+      requirementProps: {},
+      fetchingRequirementProps: false
     };
   },
   computed: {
-    disabled: function () {
-      return this.doc.hasErrors;
+    loaded() {
+      return this.mounted && this.beforeCreateFinished;
+    },
+    disabled() {
+      return this.doc.hasErrors || !!this.beforeSubmitRequirements.find(requirement => !requirement.done);
+    },
+    beforeSubmitRequirements() {
+      return this.requirements.filter(requirement => requirement.phase === 'beforeSubmit');
+    },
+    activeRequirement() {
```

Comment:
The name here is confusing. these are `afterSubmit` requirements one at a time, yeah?  I'm confused why the `find` method isn't looking for the specific phase name via `this.phase`.

Reply:
Both `afterSubmit` and `afterPasswordVerified` components are displayed with the "one at a time, sequentially fading up" presentation. Only the `beforeSubmit` components are displayed (or, well, allowed to be present) while the form is up. The difference between `afterSubmit` and `afterPasswordVerified` is that the actual initial password verification API call is made after the `afterSubmit` requirements have been handled. We have both of these because some are intended to reduce denial of service pressure on the login API, while others are intended as a second factor of authentication and only make sense after the user's identity is provisionally confirmed.

Reply:
So `activeRequirement` is appropriate because it refers to the current singular, presented-on-its-own requirement, which might be from any phase but `beforeSubmit`.